### DISCUSSION TITLE
Does next js version 12.x.x or 12.2.4 have built in support for decorators?

### DISCUSSION THREAD

**Discussion by joseDaKing** [vercel/next.js#39938] [Help] [answered]
Hi, I am currently writing Next JS app in Typescript and I want to use ORM that utilizes decorators and that's why I was wondering if Next JS version 12.x.x or 12.2.4 have built in support for decorators? I think it should be stated explicitly in the docs if there built in support for decorators in the current available version of Next JS. I have tried looking this up, and I assume there is no support for decorators but I don't know if it is just for older versions.

**Comment by icyJoseph** [comment, marked as answer]
I think for decorators you need to look at the way your code is transpiled/compiled. 

With typescript on, you just need to do as instructed here, https://www.typescriptlang.org/docs/handbook/decorators.html, by adding a `tsconfig.json`, the `"experimentalDecorators": true,` option. 

And then decorators should just work. Since decorators are still not part of the ECMA standard, it is really up to the compiler to digest them into standard code.

I found a bit more here: https://nextjs.org/docs/advanced-features/compiler#legacy-decorators

**Reply by joseDaKing** [reply]
I sincerely thank you for this, I had hard time finding this info
